src/Features/CustomerEffortScoreTracks.php

```diff
@@ -69,27 +84,61 @@ private function enable_survey_enqueing_if_tracking_is_enabled() {
 			return;
 		}
 
-		add_action(
-			'admin_init',
-			array(
-				$this,
-				'maybe_clear_ces_tracks_queue',
-			)
-		);
-
-		add_action(
-			'woocommerce_update_options',
-			array(
-				$this,
-				'run_on_update_options',
-			),
-			10,
-			3
-		);
+		add_action( 'admin_init', array( $this, 'maybe_clear_ces_tracks_queue' ) );
+		add_action( 'woocommerce_update_options', array( $this, 'run_on_update_options' ), 10, 3 );
+		add_action( 'product_cat_add_form', array( $this, 'add_script_track_product_categories' ), 10, 3 );
+		add_action( 'product_tag_add_form', array( $this, 'add_script_track_product_tags' ), 10, 3 );
+		add_action( 'woocommerce_attribute_added', array( $this, 'run_on_add_product_attributes' ), 10, 3 );
 
 		$this->onsubmit_label = __( 'Thank you for your feedback!', 'woocommerce-admin' );
 	}
 
+	/**
+	 * Returns a generated script for tracking tags added on edit-tags.php page.
+	 * CES survey is triggered via direct access to wc/customer-effort-score store
+	 * via wp.data.dispatch method.
+	 *
+	 * Due to lack of options to directly hook ourselves into the ajax post request
+	 * initiated by edit-tags.php page, we infer a successful request by observing
+	 * an increase of the number of rows in tags table
+	 *
+	 * @param string $action Action name for the survey.
+	 * @param string $label Label for the snackbar.
+	 * @param string $page_now Value of window.pagenow.
+	 * @param string $admin_page Value of window.adminpage.
+	 *
+	 * @return string Generated JavaScript to append to page.
+	 */
+	private function get_script_track_edit_php( $action, $label, $page_now, $admin_page ) {
+		return sprintf(
+			"(function( $ ) {
+				'use strict';
+				// Hook on submit button and sets a 500ms interval function
+				// to determine successful add tag or otherwise.
+				$('#addtag #submit').on( 'click', function() {
+					const initialCount = $('.tags tbody > tr').length;
+					const interval = setInterval( function() {
+						if ( $('.tags tbody > tr').length > initialCount ) {
+							// New tag detected.
+							clearInterval( interval );
+							wp.data.dispatch('wc/customer-effort-score').addCesSurvey( '%s', '%s', '%s', '%s', '%s' );
+						} else {
+							// Form is no longer loading, most likely failed.
+							if ( $( '#addtag .submit .spinner.is-active' ).length < 1 ) {
+								clearInterval( interval );
+							}
+						}
+					}, 500 );
+				});
+			})( jQuery );",
+			addslashes( $action ),
```

Comment:
We could use [esc_js](https://developer.wordpress.org/reference/functions/esc_js/) instead of `addslashes`, since it handles a few more cases that could break strings.

It's not a blocker but was there a reason not to get these strings directly from JS, for example with `window.pagenow`?

Reply:
Ahh nice catch, thanks! Changes in bac5746